File: src/vibe_kanban_clone/schemas/project.py
"""Project schemas."""

from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field


class ProjectCreate(BaseModel):
    """Schema for creating a project."""

    name: str = Field(min_length=1, max_length=255)
    description: str | None = Field(default=None, min_length=1, max_length=5000)


class ProjectRead(BaseModel):
    """Schema for reading a project."""

    model_config = ConfigDict(from_attributes=True)

    id: int
    name: str
    slug: str
    description: str | None
    created_at: datetime
    updated_at: datetime


class ProjectUpdate(BaseModel):
    """Schema for updating a project."""

    name: str | None = Field(default=None, min_length=1, max_length=255)
    description: str | None = Field(default=None, min_length=1, max_length=5000)

File: server/src/schemas/project.ts
import { z } from "zod";

export const ProjectCreate = z.object({
  name: z.string().min(1).max(255),
  description: z.string().max(5000).optional(),
});

export const ProjectRead = z.object({
  id: z.number(),
  name: z.string(),
  slug: z.string(),
  description: z.string().nullable(),
  created_at: z.string().or(z.date()),
  updated_at: z.string().or(z.date()),
});

export const ProjectUpdate = z.object({
  name: z.string().min(1).max(255).optional(),
  description: z.string().max(5000).optional(),
});

export type ProjectCreate = z.infer<typeof ProjectCreate>;
export type ProjectRead = z.infer<typeof ProjectRead>;
export type ProjectUpdate = z.infer<typeof ProjectUpdate>;
